### PR TITLE
Add more Decimal128 tests

### PR DESCRIPTION
**Pull request by bjori**
libbson currently fails the decimal128-4.json, but passes all the others :]

### CODE REVIEW

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          "description": " [dqbsr434] check rounding modes heeded (Rounded & Inexact)",
+          "subject": "1.11111111111111111111111111111234650"
+       },
+       {
+          "description": " [dqbas938] overflow results at different rounding modes (Overflow & Inexact & Rounded)",
+          "subject": "7e10000"
+       },
+       {
+          "description": "Inexct rounding#1",
+          "string": "100000000000000000000000000000000000000000000000000000000001",
```
Comment (behackett) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json]: Errant ','

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+       },
+       {
+          "description": "Inexct rounding#2",
+          "string": "1E-6177",
```
Comment (behackett) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json]: Errant ','

Comment (bjori) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json]: both fixed, thanks!

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json (hunk 1 above)
Comment (behackett) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json]: string -> subject

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json (hunk 2 above)
Comment (behackett) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json]: string -> subject

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json (hunk 1 above)
Comment (bjori) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128-4.json]: damn you test format!
The BSON document is called "subject", while the textual representation is called "string" --- except for parseErrors _sigh_.

Thanks, fixed!

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json [hunk 3]
```diff
@@ -339,21 +339,22 @@
...
             "to_extjson": false,
             "extjson": "{\"d\" : {\"$numberDecimal\" : \"-inF\"}}"
-        }
-    ],
-    "parseErrors": [
+	},
         {
-            "description": "Too many significand digits",
-            "subject": "100000000000000000000000000000000000000000000000000000000001"
+           "description": "OK3",
```
Comment (behackett) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json:344]: Maybe call this "Clamped"?

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json [hunk 4]
```diff
@@ -339,21 +339,22 @@
...
-        },
-        {
-            "description": "Too many significand digits",
-            "subject": ".100000000000000000000000000000000000000000000000000000000000"
-        },
+           "description": "OK5",
```
Comment (behackett) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json:350]: Maybe call this "Exact rounding"?

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json (hunk 3 above)
Comment (bjori) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json:344]: done

File: src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json (hunk 4 above)
Comment (bjori) [inline src/MongoDB.Bson.Tests/Specifications/bson/tests/decimal128.json:350]: done

**Comment by bjori** [comment]
This has now been pushed upstream https://github.com/10gen/specifications/commit/7e990221518553eda25b53b77662d3b4ee9f6bd6

**Pull request closed without merging**